### ISSUE TITLE
Basic support for Python 3.11's typeshed

### ISSUE THREAD

**Issue by hauntsaninja** [python/mypy#12220]
Currently, running mypy with Python 3.11 on typeshed fails. We should fix it, ideally before the next mypy release.

These should help:
https://github.com/python/typeshed/pull/7314
https://github.com/python/typeshed/pull/7318

Looks like some generic type stuff is still broken:
```
stdlib/builtins.pyi:1710: error: Value of type "Type[type]" is not indexable  [index]
```

There are changes to use `@property` in enum that mypy doesn't like. Lucky for us @sobolevn is really good at fixing enum issues :D
```
stdlib/enum.pyi:103: error: Decorated property not supported  [misc]
stdlib/enum.pyi:105: error: Decorated property not supported  [misc]
stdlib/enum.pyi:131: error: Decorated property not supported  [misc]
stdlib/enum.pyi:148: error: Signature of "value" incompatible with supertype "Flag"  [override]
stdlib/enum.pyi:159: error: Decorated property not supported  [misc]
stdlib/enum.pyi:161: error: Decorated property not supported  [misc]
stdlib/enum.pyi:189: error: Decorated property not supported  [misc]
```

**Comment by hauntsaninja** [comment]
To repro:
```
diff --git a/tests/mypy_test.py b/tests/mypy_test.py
index 4d2c77a6..f243b59d 100755
--- a/tests/mypy_test.py
+++ b/tests/mypy_test.py
@@ -285,7 +285,7 @@ def test_third_party_distribution(distribution: str, major: int, minor: int, arg
 def main():
     args = parser.parse_args()
 
-    versions = [(3, 10), (3, 9), (3, 8), (3, 7), (3, 6), (2, 7)]
+    versions = [(3, 11), (3, 10), (3, 9), (3, 8), (3, 7), (3, 6), (2, 7)]
     if args.python_version:
         versions = [v for v in versions if any(("%d.%d" % v).startswith(av) for av in args.python_version)]
         if not versions:
```
and then `python3.11 ~/dev/typeshed/tests/mypy_test.py  --python-version 3.11`

**Comment by sobolevn** [comment]
I have some crazy stuff going on, so I am not really active right now.
But, I will have a look somewhere this week! 👍 

Thanks for the repro!

**Comment by AlexWaygood** [comment]
(Note that the changes in `enum` are that `enum` in 3.11 started using `@enum.property` (not `@builtins.property`!))

**Comment by AlexWaygood** [comment]
https://github.com/python/typeshed/pull/7389 gets rid of most of the errors reported by mypy.

**Comment by AlexWaygood** [comment]
I've filed https://github.com/python/mypy/issues/12258 to track the last remaining error when running mypy's master branch on the 3.11 typeshed stubs.

**Comment by AlexWaygood** [comment]
Now that https://github.com/python/mypy/pull/12260 is merged, it looks like typeshed's 3.11 stdlib passes a mypy type-check when using the mypy master branch 🥳

**Comment by hauntsaninja** [comment]
Awesome, thanks both!
